MapboxNavigationTests/SpeechSynthesizersControllerTests.swift

```diff
@@ -0,0 +1,145 @@
+            FailingSpeechSynthesizerMock()
+        ]
+    }
+
+    override func tearDown() {
+        synthesizers = []
+    }
+
+    func testNoFallback() {
+        let speakExpectation = XCTestExpectation(description: "Priority Synthesizer should be called")
+        let dontSpeakExpectation = XCTestExpectation(description: "Fallback Synthesizer should not be called")
+        dontSpeakExpectation.isInverted = true
+        
+        (synthesizers[0] as! FailingSpeechSynthesizerMock).speakExpectation = speakExpectation
+        (synthesizers[1] as! FailingSpeechSynthesizerMock).speakExpectation = dontSpeakExpectation
+        let speechSynthesizersController = MultiplexedSpeechSynthesizer(synthesizers)
+        
+        
+        speechSynthesizersController.speak(SpokenInstruction(distanceAlongStep: .init(),
+                                                             text: "text",
+                                                             ssmlText: "text"),
+                                           during: Fixture.routeLegProgress())
+        
+        wait(for: [speakExpectation, dontSpeakExpectation], timeout: 2)
+    }
+    
+    func testFallback() {
+        let speechSynthesizersController = MultiplexedSpeechSynthesizer(synthesizers)
+        let expectation = XCTestExpectation(description: "Both Synthesizers should be called")
+        expectation.expectedFulfillmentCount = 2
+        (synthesizers[0] as! FailingSpeechSynthesizerMock).failing = true
+        (synthesizers[0] as! FailingSpeechSynthesizerMock).speakExpectation = expectation
+        (synthesizers[1] as! FailingSpeechSynthesizerMock).speakExpectation = expectation
+        
+        speechSynthesizersController.speak(SpokenInstruction(distanceAlongStep: .init(),
+                                                             text: "text",
+                                                             ssmlText: "text"),
+                                           during: Fixture.routeLegProgress())
+        
+        wait(for: [expectation], timeout: 3)
+    }
+
+    func testDeinit() {
+        let deinitExpectation = expectation(description: "Speech synthesizers should deinitialize")
+        deinitExpectation.expectedFulfillmentCount = 2
+        (synthesizers[0] as! FailingSpeechSynthesizerMock).deinitExpectation = deinitExpectation
+        (synthesizers[1] as! FailingSpeechSynthesizerMock).deinitExpectation = deinitExpectation
+        let dummyService = MapboxNavigationService(route: route, routeOptions: routeOptions)
+        
+        var routeController: RouteVoiceController? = RouteVoiceController(navigationService: dummyService,
+                                                                          speechSynthesizer: MultiplexedSpeechSynthesizer(synthesizers))
+        
+        synthesizers = []
+        routeController = nil
+        
+        wait(for: [deinitExpectation], timeout: 3)
+    }
+    
+    func testSystemSpeechSynthesizer() {
```

Comment:
Not critical, but maybe it'd be good to also verify whether properties like `muted`, `isSpeaking` etc were set correctly as well in test?

Reply:
Added a few more tests for parameters, but unfortunately could not find a stable way to test all of them. For example `isSpeaking` does not have a dedicated event after it can be "guaranteed to talk". Or `volume` value for `SystemSpeechSynthesizer` relies on system values and is not controlled by `Multiplexed` synth.